File: src/main/java/net/projecttl/website/routes/ViewRoutes.java
package net.projecttl.website.routes;

import net.projecttl.website.service.HistoryService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.GetMapping;

@Controller("/")
public class ViewRoutes {
    private final HistoryService historyService;

    @Autowired
    public ViewRoutes(HistoryService historyService) {
        this.historyService = historyService;
    }

    @GetMapping
    public String index() {
        System.out.println(historyService.hello());
        return "index";
    }

    @GetMapping("/history")
    public String history(Model model) {
        return "history";
    }
}

File: src/main/kotlin/net/projecttl/website/routes/ViewRoutes.kt
package net.projecttl.website.routes

import io.ktor.server.application.*
import io.ktor.server.response.*
import io.ktor.server.routing.*
import io.ktor.server.thymeleaf.*
import org.thymeleaf.templateresolver.ClassLoaderTemplateResolver

fun Application.viewRoutes() {
	install(Thymeleaf) {
		setTemplateResolver(ClassLoaderTemplateResolver().apply {
			prefix = "templates/thymeleaf/"
			suffix = ".html"
			characterEncoding = "utf-8"
		})
	}

	routing {
		get("/") {
			call.respond(ThymeleafContent("index", mapOf()))
		}

		get("/skills") {
			call.respond(ThymeleafContent("skills", mapOf()))
		}

		get("/history") {
			call.respond(ThymeleafContent("history", mapOf()))
		}

		get("/api") {
			call.respond(ThymeleafContent("api", mapOf()))
		}
	}
}
